Sources/SwiftParserDiagnostics/SyntaxExtensions.swift

```diff
@@ -85,7 +85,7 @@ extension Syntax {
 extension SyntaxProtocol {
   /// A name that can be used to describe this node's type in diagnostics or `nil` if there is no good name for this node.
   /// If `allowBlockNames` is `false`, ``CodeBlockSyntax`` and ``MemberDeclBlockSyntax`` are not considered to have a good name and will return `nil`.
-  func nodeTypeNameForDiagnostics(allowBlockNames: Bool) -> String? {
+  public func nodeTypeNameForDiagnostics(allowBlockNames: Bool) -> String? {
```

Comment:
I don’t think we should make this method public. The entire logic with `allowBlockNames` is very specific to `SwiftParserDiagnsotic`. Using `SyntaxKind.nameForDiagnostics` would probably be the better choice.

Reply:
> Using SyntaxKind.nameForDiagnostics would probably be the better choice.

How to make `nameForDiagnostics` public? I found the file is generated.

Reply:
The files are being generated as described here:

https://github.com/apple/swift-syntax/blob/main/CONTRIBUTING.md#generating-source-code

We need to commit the generated files to resolve a circular dependency: The sources are being generated using swift-syntax itself, which means that you need a version of swift-syntax to generate the sources of swift-syntax. If the generated sources weren’t committed anywhere, you couldn’t build swift-syntax.